src/data/upload.ts

```diff
@@ -57,17 +64,29 @@ export function queue(data: Token[]): void {
             case Event.ScriptError:
             case Event.ImageError:
                 break;
+            case Event.Backup:
+                transmit = false;
+                container = backup;
+                break;
+            case Event.Upgrade:
+                for (let entry of backup) {
```

Comment:
What is this logic doing?

Reply:
Explained in comments. It looks up backed up events and gets them ready to be sent to the server. This allows transition from lean mode into full mode.

---
In reply to: [368681147](https://github.com/microsoft/clarity-js/pull/166#discussion_r368681147) [](ancestors = 368681147)